cmd/local-gadget/snapshot/socket.go

```diff
@@ -15,54 +15,79 @@
 package snapshot
 
 import (
+	"fmt"
+
 	"github.com/spf13/cobra"
 
 	commonsnapshot "github.com/kinvolk/inspektor-gadget/cmd/common/snapshot"
-	commonutils "github.com/kinvolk/inspektor-gadget/cmd/common/utils"
 	"github.com/kinvolk/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/snapshot/socket/tracer"
+	socketTypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/snapshot/socket/types"
+	localgadgetmanager "github.com/kinvolk/inspektor-gadget/pkg/local-gadget-manager"
 )
 
 func newSocketCmd() *cobra.Command {
-	var socketFlags commonsnapshot.SocketFlags
-
-	commonFlags := &utils.CommonFlags{
-		OutputConfig: commonutils.OutputConfig{
-			// The columns that will be used in case the user does not specify
-			// which specific columns they want to print. Notice they may be
-			// extended based on flags.
-			CustomColumns: []string{
-				"pod",
-				"protocol",
-				"local",
-				"remote",
-				"status",
+	var commonFlags utils.CommonFlags
+	var flags commonsnapshot.SocketFlags
+
+	runCmd := func(*cobra.Command, []string) error {
+		socketGadget := &SnapshotGadget[socketTypes.Event]{
+			SnapshotGadgetBase: commonsnapshot.SnapshotGadgetBase[socketTypes.Event]{
+				Parser: commonsnapshot.NewSocketParserWithRuntimeInfo(&commonFlags.OutputConfig, &flags),
 			},
-		},
-	}
+			commonFlags: &commonFlags,
+			runTracer: func(localGadgetManager *localgadgetmanager.LocalGadgetManager, containerSelector *containercollection.ContainerSelector) ([]socketTypes.Event, error) {
+				allSockets := []socketTypes.Event{}
 
-	availableColumns := []string{
-		"pod",
-		"protocol",
-		"local",
-		"remote",
-		"status",
-		"inode",
-	}
+				// Given that the tracer works per network namespace, we only
+				// need to run it once per namespace.
+				visitedNetNs := make(map[string]struct{})
```

Comment:
Can you use uint14 as the map key?

Reply:
You are right. There is no reason to transform it into a string.

Beyond this, I think it is a pity to have this gadget disabled, everything is in place to add `local-gadget snapshot socket`, but we need to solve https://github.com/kinvolk/inspektor-gadget/issues/744. Maybe @alban is interested in working on it?